core/src/main/java/org/elasticsearch/index/store/StoreFileMetaData.java

```diff
@@ -38,6 +36,8 @@
 
     public static final Version FIRST_LUCENE_CHECKSUM_VERSION = Version.LUCENE_5_0_0;
 
+    public static final String UNKNOWN_CHECKSUM = "_na_";
```

Comment:
I see what you were saying now, i'll move this to `BlobStoreIndexShardSnapshot`, esp since I no longer need it for hashAndLengthEqual as its going away